systemd/datadog_checks/systemd/systemd.py

```diff
@@ -0,0 +1,175 @@
+            # We check status changes between the previous cached unit status and the current unit status
+            current_status = current_unit_status.get(unit_short_name)
+            if current_status:
+                if current_status != previous_unit_status:
+                    self.log.debug("unit {} changed state, it is now {}".format(unit_short_name, current_status))
+                    changed[unit_short_name] = current_status
+            else:
+                self.log.debug("unit {} not found".format(unit_short_name))
+                # We list all previous cached unit status that do not exist anymore
+                deleted[unit_short_name] = previous_unit_status
+
+        return changed, created, deleted
+
+    def report_changed_units(self, units, tags):
+        for unit, state in iteritems(units):
+            self.event({
+                "event_type": "unit.status.changed",
+                "msg_title": "unit {} changed state".format(unit),
+                "msg_text": "it is now: {}".format(state),
+                "tags": tags
+            })
+
+    def report_deleted_units(self, units, tags):
+        for unit, state in iteritems(units):
+            self.event({
+                "event_type": "unit.status.deleted",
+                "msg_title": "unit {} cannot be found".format(unit),  # TODO: check wording
+                "msg_text": "last reported status was: {}".format(state),
+                "tags": tags
+            })
+
+    def report_created_units(self, units, tags):
+        for unit, state in iteritems(units):
+            self.event({
+                "event_type": "unit.status.created",
+                "msg_title": "new unit {} has been found".format(unit),  # TODO: check wording
+                "msg_text": "status is: {}".format(state),
+                "tags": tags
+            })
+
+    def report_statuses(self, units, tags):
+        active_units = inactive_units = 0
+        for unit, state in iteritems(units):
+            state = ensure_unicode(state)
+            try:
+                if state == 'active':
+                    active_units += 1
+                if state == 'inactive':
+                    inactive_units += 1
+            except pystemd.dbusexc.DBusInvalidArgsError as e:
+                self.log.debug("Cannot retrieve unit status for {}, {}".format(unit, e))
+
+        self.gauge('systemd.units.active', active_units, tags)
+        self.gauge('systemd.units.inactive', inactive_units, tags)
+
+    def get_state_single_unit(self, unit_id):
+        try:
+            unit = Unit(unit_id, _autoload=True)
+            state = unit.Unit.ActiveState
+            return state
+        except pystemd.dbusexc.DBusInvalidArgsError as e:
+            self.log.info("Unit name invalid for {}, {}".format(unit_id, e))
+
+    def send_service_checks(self, unit_id, state, tags):
+        state = ensure_unicode(state)
+        if state == 'active' or 'activating':
```

Comment:
This is a nit, but maybe we can make a constant for this and failing states like `ACTIVE_STATES = ['active', 'activating']` and the same for failed states?

Reply:
To be used in `report_statuses` as well.